### ISSUE TITLE
Formulas do not respect default field values

### ISSUE THREAD

**Issue by jeffleeismyhero** [hyperoslo/Form#383]
Using the JSON below, a form us built with a select field (Ticket Type), and three numeric fields (Quantity, Price, and Total).

The Ticket Type select field properly sets the value of the Price field when the form loads based the value of the default option of the select. The Quantity field has a default value of "1."  However, the Total field is not updated until the Quantity field is manually changed.

```
[Ticket Type: 3-Day Adult Ticket ($39)]
[Quantity: 1] [Price: 39,00] [Total: 0,00]
```

The expected form should look like:

```
[Ticket Type: 3-Day Adult Ticket ($39)]
[Quantity: 1] [Price: 39,00] [Total: 39,00]
```

#### Non-working JSON (seems like this should work)
```json
{
  "groups":[
    {
      "id":"tickets",
      "title":"Tickets",
      "sections":[
        {
          "id":"tickets-0",
          "fields":[
            {
              "id":"tickettype",
              "title":"Ticket Type",
              "info":"Type of ticket",
              "type":"select",
              "size":{
                "width":100,
                "height":1
              },
              "values":[
                {
                  "id":0,
                  "title":"3-Day Adult Ticket ($39)",
                  "default":true,
                  "value":39,
                  "targets":[
                    {
                      "id":"price",
                      "type":"field",
                      "action":"update"
                    }
                  ]
                },
                {
                  "id":1,
                  "title":"3-Day Child Ticket ($10)",
                  "value":10,
                  "targets":[
                    {
                      "id":"price",
                      "type":"field",
                      "action":"update"
                    }
                  ]
                }
              ]
            },
            {
              "id":"quantity",
              "title":"Quantity",
              "type":"number",
              "value":"1",
              "size":{
                "width":25,
                "height":1
              },
              "targets":[
                {
                  "id":"total",
                  "type":"field",
                  "action":"update"
                }
              ],
              "validations":{
                "max_value":100,
                "min_value":1,
                "required":true
              }
            },
            {
              "id":"price",
              "title":"Price",
              "type":"float",
              "size":{
                "width":25,
                "height":1
              },
              "targets":[
                {
                  "id":"total",
                  "type":"field",
                  "action":"update"
                }
              ],
              "formula":"$tickettype",
              "validations":{
                "max_value":100,
                "min_value":1,
                "required":true
              }
            },
            {
              "id":"total",
              "title":"Total",
              "type":"float",
              "size":{
                "width":25,
                "height":1
              },
              "disabled":true,
              "formula":"$quantity * $price"
            }
          ]
        }
      ]
    }
  ]
}
```

#### Working JSON with formula of "1" for Quantity and quantity target on select field
```json
{
  "groups":[
    {
      "id":"tickets",
      "title":"Tickets",
      "sections":[
        {
          "id":"tickets-0",
          "fields":[
            {
              "id":"tickettype",
              "title":"Ticket Type",
              "info":"Type of ticket",
              "type":"select",
              "size":{
                "width":100,
                "height":1
              },
              "values":[
                {
                  "id":0,
                  "title":"3-Day Adult Ticket ($39)",
                  "default":true,
                  "value":39,
                  "targets":[
                    {
                      "id":"price",
                      "type":"field",
                      "action":"update"
                    },
                    {
                      "id":"quantity",
                      "type":"field",
                      "action":"update"
                    }
                  ]
                },
                {
                  "id":1,
                  "title":"3-Day Child Ticket ($10)",
                  "value":10,
                  "targets":[
                    {
                      "id":"price",
                      "type":"field",
                      "action":"update"
                    },
                    {
                      "id":"quantity",
                      "type":"field",
                      "action":"update"
                    }
                  ]
                }
              ]
            },
            {
              "id":"quantity",
              "title":"Quantity",
              "type":"number",
              "value":"1",
              "size":{
                "width":25,
                "height":1
              },
              "targets":[
                {
                  "id":"total",
                  "type":"field",
                  "action":"update"
                }
              ],
              "validations":{
                "max_value":100,
                "min_value":1,
                "required":true
              },
              "formula":"1"
            },
            {
              "id":"price",
              "title":"Price",
              "type":"float",
              "size":{
                "width":25,
                "height":1
              },
              "targets":[
                {
                  "id":"total",
                  "type":"field",
                  "action":"update"
                }
              ],
              "formula":"$tickettype",
              "validations":{
                "max_value":100,
                "min_value":1,
                "required":true
              }
            },
            {
              "id":"total",
              "title":"Total",
              "type":"float",
              "size":{
                "width":25,
                "height":1
              },
              "disabled":true,
              "formula":"$quantity * $price"
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by 3lvis** [comment]
Thanks for reporting this @jeffleeismyhero! Will look into it.

**Comment by 3lvis** [comment]
Hey @jeffleeismyhero, I was able to reproduce the issue. We haven't faced this one because we use the values parameter to do this kind of things in our apps.

```objc
dataSource = [[FORMDataSource alloc] initWithJSON:self.JSON
                                   collectionView:self.collectionView
                                           layout:self.layout
                                           values:@{@"quantity" : @1}
                                         disabled:YES];
```

**Comment by 3lvis** [comment]
Fixed in https://github.com/hyperoslo/Form/pull/384

**Comment by 3lvis** [comment]
Hi @jeffleeismyhero! This issue is fixed on 3.3.5, thanks!

https://github.com/hyperoslo/Form/releases/tag/3.3.5

**Comment by jeffleeismyhero** [comment]
Thanks, that seems to work fine now.
